CHANGELOG.md

```diff
@@ -111,7 +111,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated `DataModule` properties: `train_transforms`, `val_transforms`, `test_transforms`, `size`, `dims` ([#8851](https://github.com/PyTorchLightning/pytorch-lightning/pull/8851))
 
 
--
+- Deprecated `TrainingTypePlugin.on_save` and `Accelerator.on_save` ([#8987](https://github.com/PyTorchLightning/pytorch-lightning/pull/8987))
```

Comment:
this would be a removal as we're deleting the functionality altogether. the typical procedure is to mark the API as deprecated, and then make a breaking change a few versions later. if this approach is okay, we should move this line to the Removed section below

Reply:
Thank you for pointing it out @ananthsub ! What's your recommendation for the correct approach? 
I feel the save_checkpoint() is already a replacement for on_save(). The on_save() function just return the param it got, so I think it's safe to remove it directly. What's your advise on this? Should we always deprecate first then remove in next few versions?

Reply:
I agree. Let's add this directly to the `Removed` section